docs_src/microservices/security/Ch-Docker-Swarm-SecuringDeviceServices.md

```diff
@@ -0,0 +1,218 @@
+# Security for EdgeX Stack
+
+This page shows how to secure communication between core EdgeX services and various device services by utilizing docker swarm to create an encrypted overlay network between two hosts.  We are showcasing two interesting concepts here.  
+
+1) Securing the traffic between core and device services
+2) Setting up an EdgeX Stack cross platform using docker swarm
+
+## Docker Swarm Overlay Network
+
+Docker's overlay network driver is a software abstraction on top of physical networking hardware to link multiple nodes together in a distributed network. This allows nodes/containers running on the network to communicate securely, if encryption in enabled.
+
+We created two docker swarm nodes for this example a manager node and a worker node.  The manager node is running all of the core EdgeX services and the worker node runs the device services.  Using the docker daemon's overlay network abstraction and enabling security we can have secure communication between these nodes.
+
+![image](overlay-network.png)
+
+
+## Reference implementation example
+
+The reference implementation example can be found in this repository:
+[Reference example device-service docker-swarm overlay network](https://add-repo/README.md)
+
+### Setup remote running Virtual Machine
+
+In this example setup, similar to the the [SSH example](https://github.com/edgexfoundry/edgex-docs/blob/master/docs_src/microservices/security/Ch-SSH-Tunneling-HowToSecureDeviceServices.md), `vagrant` is used on the top of `Virtual Box` to set up the secondary/remote VM.
+
+Download vagrant from Hashicorp website or if you're on Ubuntu via `sudo apt install virtualbox` and `sudo apt install vagrant`.  We have a simple vagrant file used for this tutorial [here](https://github.com/edgexfoundry/edgex-examples/blob/swarm/security/docker-swarm/ssh-tunneling/Vagrantfile)
+
+This vagrant file sets the hostname for the new VM and installs docker.
+
+### Getting the VM running
+
+- Launch the worker node or VM if it is not yet:
+
+```sh
+vagrant up
+```
+
+ssh into the worker node via 
+
+```sh
+vagrant ssh
+```
+
+This will give you a terminal prompt in the worker node where you will run the `sudo docker swarm join` command in a few steps.
+
+### Connecting the swarm nodes
+
+With the VM up and running we need to connect the two nodes using docker swarm.
+
+The following command initializes a docker swarm and is to be ran on the host machine: 
+
+```sh
+sudo docker swarm init --advertise-addr <your ip address>
```

Comment:
Is this the ip address of the host or VM?

Reply:
Got point I'll update that it's the host's IP